starter-code/tests/FunctionsAndArraysSpec.js

```diff
@@ -1,246 +1,418 @@
+//     var matrix = [
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2],
+//       [2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2]
+//     ];
+//     expect(greatestProduct(matrix)).toBe(16);
+//   });
+// });
+
+
+
+
+
+
+
+
+
+
+// ---------  My Answers Below ------------------------------------------------------------------------------
+
+// David Mudd & Ali Sattar
+
+// function maxOfTwoNumbers(num1, num2) {
+//   if (num1 > num2) {
+//     console.log(`${num1} is larger than ${num2}.`);
+//   } else {
+//     console.log(`${num2} is larger than ${num1}.`)
+//   }
+// }
+
+// maxOfTwoNumbers(12,11);
+
+// --------------------------
+
+// var words = [
+//   'mystery',
+//   'brother',
+//   'aviator',
+//   'crocodile',
+//   'pearl',
+//   'orchard',
+//   'crackpot'
+// ];
+
+
+// var longest = "";
+
+// words.forEach(function(oneWord) {
+// if(longest.length < oneWord.length){
+//   longest = oneWord
+// }
+// return longest
+// });
+// console.log("longest is: ", longest);
+
+// ------------------------------
+
+// var numbers = [6, 12, 1, 18, 13, 16, 2, 1, 8, 10];
+
+
+// function calculateSum(arrayOfNumbers) {
+
+//   var sum = 0;
+
+//   arrayOfNumbers.forEach(function(oneNumber) {
+//    sum += oneNumber;
+//   });
+
+//   return sum;
+// };
+
+// console.log(calculateSum(numbers));
+
+
+// ------------------------
 
+// var numbers = [2, 6, 9, 10, 7, 4, 1, 9];
+
+// // function calcAvg(numberArray) {
+// //   var sum = 0;
+// // }
+
+
+// var sum = 0;
+// function sumBlah(array){
+//   array.forEach(function(placeholder) {
+//     sum += placeholder;
+//   });
+//     return sum;
+// }
+
+// // sumBlah(numbers)
```

Comment:
calling this first and then again within the function below having a global variable without a reset will give you an incorrect result.